sdk/core/azure_core_amqp/src/messaging.rs

```diff
@@ -1075,74 +1105,95 @@ impl AmqpMessage {
         self.footer.as_ref()
     }
 
-    pub fn set_header(&mut self, header: AmqpMessageHeader) {
-        self.header = Some(header);
-    }
-
-    pub fn set_properties(&mut self, properties: AmqpMessageProperties) {
-        self.properties = Some(properties);
-    }
-
-    pub fn set_message_annotations(&mut self, message_annotations: impl Into<AmqpAnnotations>) {
-        self.message_annotations = Some(message_annotations.into());
-    }
-
-    pub fn set_message_body(&mut self, body: impl Into<AmqpMessageBody>) {
-        self.body = body.into();
+    /// Set the message ID for the message.
+    ///
+    /// This will overwrite any existing message ID on the message.
+    ///
+    /// # Arguments
+    ///
+    /// * `message_id` - The message ID to set on the message.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use azure_core_amqp::messaging::AmqpMessage;
+    ///
+    /// let mut message = AmqpMessage::default();
+    /// message.set_message_id(uuid::Uuid::new_v4());
+    /// message.set_message_id("This is a message ID");
+    /// ```
+    ///
+    pub fn set_message_id(&mut self, message_id: impl Into<AmqpMessageId>) {
+        if let Some(properties) = self.properties.as_mut() {
+            properties.set_message_id(message_id);
+            self.properties = Some(properties.clone());
+        } else {
+            self.properties = Some(
+                AmqpMessageProperties::builder()
+                    .with_message_id(message_id)
+                    .build(),
+            );
+        }
     }
 
-    pub fn set_application_properties(
+    /// Adds a message annotation for the message.
+    /// If the message annotations already exist, the annotation will be added to the existing annotations.
```

Comment:
You should separate this with a new line. Much like git commit messages, the first line is the API "summary" and the subsequent text is the description / more info. But since it's markdown, if you don't have a blank line, it's all one paragraph.

```suggestion
    /// Adds a message annotation for the message.
    ///
    /// If the message annotations already exist, the annotation will be added to the existing annotations.
```